jaxrs/src/main/java/com/facebook/airlift/jaxrs/JerseyAuthorizationFilter.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2010 Proofpoint, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.airlift.jaxrs;
+
+import com.facebook.airlift.http.server.AuthorizationResult;
+import com.facebook.airlift.http.server.Authorizer;
+import com.facebook.airlift.http.server.HttpServerConfig;
+import com.google.common.collect.ImmutableSet;
+import com.google.inject.Inject;
+
+import javax.annotation.security.RolesAllowed;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerRequestFilter;
+import javax.ws.rs.container.ResourceInfo;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.Provider;
+
+import java.security.Principal;
+import java.util.Optional;
+import java.util.Set;
+
+import static java.lang.String.format;
+
+@Provider
+public class JerseyAuthorizationFilter
+        implements ContainerRequestFilter
+{
+    private final Authorizer authorizer;
+    private final HttpServerConfig httpServerConfig;
+
+    @Context
+    private ResourceInfo resourceInfo;
+
+    @Inject
+    public JerseyAuthorizationFilter(Authorizer authorizer, HttpServerConfig httpServerConfig)
+    {
+        this.authorizer = authorizer;
```

Comment:
Presto has this convention of adding `requireNonNull` in constructor assignments. You can grep for usage style for this in the codebase.